src/test/util_tests.cpp

```diff
@@ -78,6 +78,29 @@ BOOST_AUTO_TEST_CASE(util_datadir)
     BOOST_CHECK_EQUAL(dd_norm, args.GetDataDirBase());
 }
 
+namespace {
+class NoCopyOrMove
+{
+public:
+    int i = 0;
+    explicit NoCopyOrMove(int i) : i{i} { }
+    NoCopyOrMove(const NoCopyOrMove&) = delete;
+    NoCopyOrMove(NoCopyOrMove&&) = delete;
+    NoCopyOrMove& operator=(const NoCopyOrMove&) = delete;
+    NoCopyOrMove& operator=(NoCopyOrMove&&) = delete;
+
+    operator bool() const { return i != 0; }
+
+    int getp1() { return i + 1; }
+    int test() {
+        // Check that Assume can be used within a lambda and still call methods
+        [&]() { Assume(getp1()); }();
+        if (Assume(getp1() != 5)) return 3;
+        return 9;
```

Comment:
Looks like it would be sufficient to return a `bool` with either `true` or `false`, here?